src/ripple/legacy/0.27/CreateOffer.cpp

```diff
@@ -0,0 +1,631 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/legacy/0.27/book/OfferStream.h>
+#include <ripple/legacy/0.27/book/Taker.h>
+#include <ripple/legacy/0.27/book/Types.h>
+#include <ripple/legacy/0.27/book/Amounts.h>
+#include <ripple/legacy/0.27/CreateOffer.h>
+#include <ripple/basics/Log.h>
+#include <ripple/core/Config.h>
+#include <ripple/json/to_string.h>
+#include <beast/cxx14/memory.h>
+
+namespace ripple {
+
+namespace legacy {
+
+TER
+CreateOffer::checkAcceptAsset(IssueRef issue) const
+{
+    /* Only valid for custom currencies */
+    assert (!isXRP (issue.currency));
+
+    SLE::pointer const issuerAccount = mEngine->entryCache (
+        ltACCOUNT_ROOT, getAccountRootIndex (issue.account));
+
+    if (!issuerAccount)
+    {
+        if (m_journal.warning) m_journal.warning <<
+            "delay: can't receive IOUs from non-existent issuer: " <<
+            to_string (issue.account);
+
+        return (mParams & tapRETRY)
+            ? terNO_ACCOUNT
+            : tecNO_ISSUER;
+    }
+
+    if (issuerAccount->getFieldU32 (sfFlags) & lsfRequireAuth)
+    {
+        SLE::pointer const trustLine (mEngine->entryCache (
+            ltRIPPLE_STATE, getRippleStateIndex (
+                mTxnAccountID, issue.account, issue.currency)));
+
+        if (!trustLine)
+        {
+            return (mParams & tapRETRY)
```

Comment:
This is the code, exactly as it exists in the 0.27.x branch.I didn't want to modify it anymore than I absolutely had to.